testkit/src/builder.rs

```diff
@@ -111,6 +111,8 @@ pub struct TestKitBuilder {
     validator_count: Option<u16>,
     service_instances: Vec<InstanceCollection>,
     logger: bool,
+    runtime_factories: Vec<Box<dyn RuntimeFactory>>,
+    instances: Vec<InstanceConfig>,
```

Comment:
So where is the difference between `service_instances` and `instances`? Why such duplication is needed?